include/solvers/navier_stokes_scratch_data.h

```diff
@@ -268,6 +280,11 @@ class NavierStokesScratchData : public PhysicsScratchDataBase
       compute_cell_measure_with_JxW(this->fe_values.get_JxW_values());
     this->cell_size = compute_cell_diameter<dim>(cell_measure, fe.degree);
 
+    // For the SDIRK methods, \sum_{j=1}^{i-1} a_{ij} k_j is needed for the
+    // assembler
+    this->fe_values[velocities].get_function_values(sum_over_previous_stages,
+                                                    this->sdirk_stage_sum);
```

Comment:
You have access to the simulation control right? I wonder if  we could get_function_values of this field only and only if the method is sdirk? This would save additional calculations.